src/plugins/discover/public/application/angular/discover.js

```diff
@@ -688,6 +666,12 @@ function discoverController(
         }
       });
 
+      $scope.changeInterval = interval => {
+        if (interval) {
+          setAppState({ interval });
+        }
+      };
+
```

Comment:
Nice, this means you can further clean up angular watchers. Line 663-667 (`$scope.$watch('state.interval',`) are no longer necessary 👍